src/main/kotlin/lottoAuto/domain/LottoRank.kt

```diff
@@ -5,12 +5,21 @@ enum class LottoRank(val matchCount: Int, val winningMoney: Int) {
     FOURTH(3, 5_000),
     THIRD(4, 50_000),
     SECOND(5, 1_500_000),
+    BONUS(5, 30_000_000),
     FIRST(6, 2_000_000_000);
 
     companion object {
-        fun from(lotto: Lotto, winningLotto: Lotto): LottoRank {
+        fun from(matchCount: Int, withBonus: Boolean = false): LottoRank {
+            when (matchCount) {
+                SECOND.matchCount -> {
+                    if (withBonus) {
+                        return BONUS
+                    }
+                    return SECOND
+                }
+            }
```

Comment:
아래와 같은 if 문이 하나 추가된다면 동일하게 로직을 처리할 수 있을 것 같아요. 🤔
```suggestion
            if (BONUS.matchCount == matchCount && withBonus) {
                return BONUS
            }
```

Reply:
굳이 3 depth 쓰지 않고 간결하게 표현하는게 맞겠네요..! [알려주신 방향으로 수정](https://github.com/next-step/kotlin-lotto/pull/955/commits/39277535917e9dd30619ce5fb5f5f2621bbd47cd)했습니다. 감사합니다!!